x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/eql/trial_license_complete_tier/eql_alert_suppression.ts

```diff
@@ -1778,40 +1785,1463 @@ export default ({ getService }: FtrProviderContext) => {
           rule,
+          timeframeEnd: new Date('2020-10-28T07:00:00.000Z'),
           invocationCount: 1,
         });
+        const previewAlerts = await getPreviewAlerts({
+          es,
+          previewId,
+          sort: [ALERT_ORIGINAL_TIME],
+        });
+        // we expect one alert and two suppressed alerts
+        // and two building block alerts, let's confirm that
+        expect(previewAlerts.length).toEqual(3);
+        const [sequenceAlert, buildingBlockAlerts] = partition(
+          previewAlerts,
+          (alert) => alert?._source?.['kibana.alert.building_block_type'] == null
+        );
+        const [suppressedSequenceAlerts] = partition(
+          sequenceAlert,
+          (alert) => (alert?._source?.['kibana.alert.suppression.docs_count'] as number) >= 0
+        );
+        expect(suppressedSequenceAlerts.length).toEqual(1);
+        expect(buildingBlockAlerts.length).toEqual(2);
+        expect(sequenceAlert.length).toEqual(1);
+
+        expect(sequenceAlert[0]?._source).toEqual({
+          ...sequenceAlert[0]?._source,
+          [ALERT_SUPPRESSION_TERMS]: [
+            {
+              field: 'host.name',
+              value: null,
+            },
+          ],
+          [TIMESTAMP]: '2020-10-28T07:00:00.000Z',
+          [ALERT_LAST_DETECTED]: '2020-10-28T07:00:00.000Z',
+          [ALERT_SUPPRESSION_DOCS_COUNT]: 2,
+        });
+      });
+
+      it('does not suppress alerts when "doNotSuppress" is set and suppression field value is undefined for a sequence alert in a given rule execution', async () => {
+        // This logic should be understood within the confines of the
+        // objectPairIntersection function defined in
+        // x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/eql/build_alert_group_from_sequence.ts
+        // Any sequence comprised of events where one field contains a value, followed by any number of
+        // events in the sequence where the value is null or undefined will have that field
+        // stripped from the sequence alert. So given that, we expect three alerts here
+        // all with the suppression value as undefined
+        const id = uuidv4();
+        const timestamp = '2020-10-28T06:50:00.000Z';
+        const laterTimestamp = '2020-10-28T06:51:00.000Z';
+        const laterTimestamp2 = '2020-10-28T06:53:00.000Z';
+        const laterTimestamp3 = '2020-10-28T06:53:01.000Z';
+
+        const doc1 = {
+          id,
+          '@timestamp': timestamp,
+          host: { name: 'host-a' },
+        };
+        const doc1WithNoHost = {
+          ...doc1,
+          '@timestamp': laterTimestamp,
+          host: { name: undefined },
+        };
+
+        const doc2WithNoHost = {
+          ...doc1,
+          '@timestamp': laterTimestamp2,
+          host: { name: undefined },
+        };
+
+        const doc3WithNoHost = {
+          ...doc1,
+          '@timestamp': laterTimestamp3,
+          host: { name: undefined },
```

Comment:
The objective for this test is exercising the `doNotSuppress` option for the `missing_fields_strategy`. I think I need `undefined` for the suppression fields in order to accomplish this.

Reply:
I could add separate values for each event in the sequence and the outcome would be the same, this is a correct observation. I'm concerned that the parity with other rule type suppression funcationlity would not be as clear as it should be - the logic for how the sequence alert acquires values for its' fields where distinct values for the same field will yield a `null` value is different from other rule types. While this will yield a similar result for this test, I feel this specific functionality is better exemplified by the test labeled `suppresses sequence alert with suppression value of "null" when all events have unique values for suppression field`

Reply:
Maybe mix of undefined and values for host names would be helpful?
It would show, alerts won't be suppressed in both cases, highlighting unique to EQL functionality